packages/edit-post/src/components/sidebar/template/index.js

```diff
@@ -43,18 +43,32 @@ export function TemplatePanel() {
 			getEditorSettings,
 			getCurrentPostType,
 		} = select( editorStore );
-		const { getPostType } = select( coreStore );
+		const { getPostType, getEntityRecords } = select( coreStore );
 		const _isViewable =
 			getPostType( getCurrentPostType() )?.viewable ?? false;
 		const _supportsTemplateMode =
 			select( editorStore ).getEditorSettings().supportsTemplateMode &&
 			_isViewable;
 
+		const templates = getEntityRecords( 'postType', 'wp_template', {
+			per_page: -1,
+		} );
+
+		const newAvailableTemplates = fromPairs(
+			( templates || [] ).map( ( { slug, title } ) => [
+				slug,
+				title.rendered,
+			] )
+		);
+
 		return {
 			isEnabled: isEditorPanelEnabled( PANEL_NAME ),
 			isOpened: isEditorPanelOpened( PANEL_NAME ),
 			selectedTemplate: getEditedPostAttribute( 'template' ),
-			availableTemplates: getEditorSettings().availableTemplates,
+			availableTemplates: {
+				...getEditorSettings().availableTemplates,
```

Comment:
I wish we could remove this entirely but it seems we can't yet because we can't retrieve the "php" custom templates using the REST API yet.